llama_index/objects/table_node_mapping.py

```diff
@@ -0,0 +1,63 @@
+"""Table node mapping."""
+
+from llama_index.objects.base_node_mapping import BaseObjectNodeMapping
+
+from typing import Any, Sequence, Optional
+from pydantic import BaseModel
+from llama_index.langchain_helpers.sql_wrapper import SQLDatabase
+from llama_index.data_structs.node import Node
+
+
+class SQLTableSchema(BaseModel):
+    """Lightweight representation of a SQL table."""
+
+    table_name: str
+    context_str: Optional[str] = None
+
+
+class SQLTableNodeMapping(BaseObjectNodeMapping[SQLTableSchema]):
+    """SQL Table node mapping."""
+
+    def __init__(self, sql_database: SQLDatabase) -> None:
+        self._sql_database = sql_database
+
+    @classmethod
+    def from_objects(
+        cls,
+        objs: Sequence[SQLTableSchema],
+        *args: Any,
+        sql_database: Optional[SQLDatabase] = None,
+        **kwargs: Any,
+    ) -> "BaseObjectNodeMapping":
+        """Initialize node mapping."""
+        if sql_database is None:
+            raise ValueError("Must provide sql_database")
+        # ignore objs, since we are building from sql_database
+        return cls(sql_database)
+
+    def _add_object(self, obj: SQLTableSchema) -> None:
+        raise NotImplementedError
+
+    def to_node(self, obj: SQLTableSchema) -> Node:
+        """To node."""
+        # taken from existing schema logic
+        table_text = (
+            f"Schema of table {obj.table_name}:\n"
+            f"{self._sql_database.get_single_table_info(obj.table_name)}\n"
+        )
+        if obj.context_str is not None:
+            table_text += f"Context of table {obj.table_name}:\n"
+            table_text += obj.context_str
+
+        return Node(
+            text=table_text,
+            node_info={"name": obj.table_name, "context": obj.context_str},
```

Comment:
Any reason for using node_info here? Technically this field will be removed soon :)

Reply:
yeah...it was the main way to store information while still giving us control over what to put in node_text.

another hacky approach is just create a JSON dict and serialize to string

Reply:
Fair enough! Thankfully, the new schema still allows this control (just slightly more verbose I suppose)